plugins/out_s3/s3.c

```diff
@@ -419,7 +419,8 @@ static int cb_s3_init(struct flb_output_instance *ins,
             flb_plg_error(ctx->ins, "Failed to parse upload_chunk_size %s", tmp);
             return -1;
         }
-        if (ctx->upload_chunk_size > ctx->file_size) {
+        if (ctx->upload_chunk_size > ctx->file_size && 
+            ctx->file_size > MIN_CHUNKED_UPLOAD_SIZE) {
```

Comment:
this is not the fix I had in mind... the logic is also wrong. PutObject should be set to true and upload_chunk_size should be ignored when `ctx->file_size < 2 * MIN_CHUNKED_UPLOAD_SIZE` is true. 

There is already code that sets `use_put_object` to true if that condition is met. That code is on line 437. What you should do is move that block outside of this larger if statement. Put the following code before line 416:


```
if (ctx->use_put_object == FLB_FALSE && ctx->file_size < 2 * MIN_CHUNKED_UPLOAD_SIZE) {
            flb_plg_info(ctx->ins,
                         "total_file_size is less than 10 MB, will use PutObject API");
            ctx->use_put_object = FLB_TRUE;
        }
```

Then the entire block of code starting on 416 will be ignored and none of the checks for multipart will run.

Reply:
Yes, I noticed that part. From my understanding, I think the code didn't go to that block is because this if statement `if (ctx->upload_chunk_size > ctx->file_size)`. Sometimes the default `upload_chunk_size` is large than the `total_file_size` we set. But if the file size is smaller than 1M, it will fail. So I just add one line here.

Reply:
Yes but if you move the block of code I mentioned farther up in the code, the logic will be fixed.

Reply:
And my suggestion is simpler and cleaner I think, since it only checks the condition `ctx->file_size < 2 * MIN_CHUNKED_UPLOAD_SIZE` once.